src/views/Users/ViewUser/index.jsx

```diff
@@ -87,53 +90,101 @@ function ViewUser({ isNewUser, ...props }) {
-  const error = userAction.error || productsAction.error || saveAction.error;
+  const [SCAction, fetchSC] = useAction(getScheduledChanges);
+  const [saveAction, saveUser] = useAction(updateUser);
+  const isLoading =
+    userAction.loading ||
+    productsAction.loading ||
+    rsAction.loading ||
+    SCAction.loading;
+  const error =
+    userAction.error ||
+    productsAction.error ||
+    saveAction.error ||
+    rsAction.error ||
+    SCAction.error;
   const defaultToEmptyString = defaultTo('');
 
   useEffect(() => {
     if (!isNewUser) {
-      Promise.all([fetchUser(existingUsername), fetchProducts()]).then(
-        ([userdata, productdata]) => {
-          const roles = Object.keys(userdata.data.data.roles).map(name => ({
-            name,
-            data_version: userdata.data.data.roles[name].data_version,
-            metadata: {
-              isAdditional: false,
-            },
-          }));
-          const permissions = Object.keys(userdata.data.data.permissions).map(
-            name => {
-              const details = userdata.data.data.permissions[name];
-
-              return {
-                name,
-                options: details.options || { products: [], actions: [] },
-                data_version: details.data_version,
-                metadata: {
-                  isAdditional: false,
-                  productText: '',
-                  actionText: '',
-                },
-              };
+      Promise.all([
+        fetchUser(existingUsername),
+        fetchProducts(),
+        fetchRS(OBJECT_NAMES.PRODUCT_REQUIRED_SIGNOFF),
+        fetchSC(),
+      ]).then(([userdata, productdata, rs, scheduledChanges]) => {
+        const roles = Object.keys(userdata.data.data.roles).map(name => ({
+          name,
+          data_version: userdata.data.data.roles[name].data_version,
+          metadata: {
+            isAdditional: false,
+          },
+        }));
+        const permissions = Object.keys(userdata.data.data.permissions).map(
+          name => {
+            const details = userdata.data.data.permissions[name];
+            const sc = scheduledChanges.data.data.scheduled_changes.filter(
+              sc => sc.permission === name
+            );
+            const permission = {
+              name,
+              options: details.options || { products: [], actions: [] },
+              data_version: details.data_version,
+              sc: null,
+              metadata: {
+                isAdditional: false,
+                productText: '',
+                actionText: '',
+              },
+            };
+
+            if (sc.length > 0) {
+              sc[0].name = sc[0].permission;
+              delete sc[0].permission;
+
+              if (
+                sc[0].options &&
+                !Object.keys(sc[0].options).includes('actions')
+              ) {
+                sc[0].options.actions = [];
+              }
+
+              [permission.sc] = sc;
```

Comment:
A comment in the code about why these previous couple of lines are necessary would make it easier to understand.

Reply:
I think the `actions` block is actually unnecessary now. I've removed it and added some comments about the other bits, though.